src/actions/tests/pending-breakpoints.js

```diff
@@ -0,0 +1,261 @@
+const theMockedPendingBreakpoint = {
+  location: {
+    sourceUrl: "http://localhost:8000/examples/bar.js",
+    line: 5,
+    column: undefined
+  },
+  condition: "3",
+  disabled: false
+};
+
+jest.mock("../../utils/prefs", () => ({
+  prefs: {
+    expressions: [],
+    pendingBreakpoints: {
+      "http://localhost:8000/examples/bar.js:5:": {
+        location: {
+          sourceUrl: "http://localhost:8000/examples/bar.js",
+          line: 5,
+          column: undefined
+        },
+        condition: "3",
+        disabled: false
+      }
+    }
+  }
+}));
+
+import {
+  createStore,
+  selectors,
+  actions,
+  makeSource
+} from "../../utils/test-head";
+import {
+  makeLocationId,
+  makePendingLocationId
+} from "../../reducers/breakpoints";
+import expect from "expect.js";
+
+function generateBreakpoint(filename) {
+  return {
+    location: {
+      sourceUrl: `http://localhost:8000/examples/${filename}`,
+      sourceId: filename,
+      line: 5
+    },
+    condition: null,
+    disabled: false
+  };
+}
+
+function generatePendingBreakpoint(breakpoint) {
+  const {
+    location: { sourceUrl, line, column },
+    condition,
+    disabled
+  } = breakpoint;
+
+  return {
+    location: { sourceUrl, line, column },
+    condition,
+    disabled
+  };
+}
+
+function slideMockBp(bp) {
+  const slidBp = Object.assign({}, bp);
+  slidBp.location.line = bp.location.line + 2;
+  return slidBp;
+}
+
+const slidingMockThreadClient = {
+  setBreakpoint: (location, condition) => {
+    return new Promise((resolve, reject) => {
+      const actualLocation = Object.assign({}, location, {
+        line: location.line + 2
+      });
+      resolve({ id: makeLocationId(location), actualLocation, condition });
+    });
+  }
+};
+
+const simpleMockThreadClient = {
+  setBreakpoint: (location, condition) => {
+    return new Promise((resolve, reject) => {
+      resolve({ id: "hi", actualLocation: location });
+    });
+  },
+
+  removeBreakpoint: id => {
+    return new Promise((resolve, reject) => {
+      resolve({ status: "done" });
+    });
+  },
+
+  setBreakpointCondition: (id, location, condition, noSliding) => {
+    return new Promise((resolve, reject) => {
+      resolve({ sourceId: "a", line: 5 });
+    });
+  }
+};
+
+describe("pending breakpoints", () => {
+  it("when the user adds a breakpoint, a corresponding pending breakpoint should be added", async () => {
+    const { dispatch, getState } = createStore(simpleMockThreadClient);
+    const bp = generateBreakpoint("foo");
+    const id = makePendingLocationId(bp.location);
+
+    await dispatch(actions.addBreakpoint(bp.location));
+    const pendingBps = selectors.getPendingBreakpoints(getState());
+    expect(pendingBps.size).to.be(2);
+    expect(pendingBps.get(id)).to.eql(generatePendingBreakpoint(bp));
+  });
+
+  it("when the user adds a sliding breakpoint, a corresponding pending breakpoint should be added", async () => {
```

Comment:
Looks good. Unlike the above two tests, this can be its own case (outside of the other two tests that have the same setup)
doesnt require a before each until we have another test with these setup conditions